python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -990,6 +990,127 @@ interface {{ ethernet_interface.name }}
       priority {{ ethernet_interface.sync_e.priority }}
 {%         endif %}
 {%     endif %}
+{%     if ethernet_interface.switchport.tap is arista.avd.defined or ethernet_interface.switchport.tool is arista.avd.defined %}
+{%         if ethernet_interface.switchport.tap.native_vlan is arista.avd.defined %}
+   switchport tap native vlan {{ ethernet_interface.switchport.tap.native_vlan }}
+{%         endif %}
+{%         if ethernet_interface.switchport.tap.identity.id is arista.avd.defined %}
+{%             set tap_identity_cli = "switchport tap identity " ~ ethernet_interface.switchport.tap.identity.id %}
+{%             if ethernet_interface.switchport.tap.identity.inner_vlan is arista.avd.defined %}
+{%                 set tap_identity_cli = tap_identity_cli ~ " inner " ~ ethernet_interface.switchport.tap.identity.inner_vlan %}
+{%             endif %}
+   {{ tap_identity_cli }}
+{%         endif %}
+{%         if ethernet_interface.switchport.tap.mac_address.dest is arista.avd.defined %}
+{%             set tap_mac_address_cli = "switchport tap mac-address dest " ~ ethernet_interface.switchport.tap.mac_address.dest %}
+{%             if ethernet_interface.switchport.tap.mac_address.src is arista.avd.defined %}
+{%                 set tap_mac_address_cli = tap_mac_address_cli ~ " src " ~ ethernet_interface.switchport.tap.mac_address.src %}
+{%             endif %}
+   {{ tap_mac_address_cli }}
+{%         endif %}
+{# The position of "mpls_pop_all" does not match the EOS CLI to implement this "elif" with "vxlan_strip" #}
+{%         if ethernet_interface.switchport.tap.mpls_pop_all is arista.avd.defined(true) %}
+   switchport tap mpls pop all
+{%         elif ethernet_interface.switchport.tap.encapsulation.vxlan_strip is arista.avd.defined(true) %}
+   switchport tap encapsulation vxlan strip
+{%         endif %}
+{%         if ethernet_interface.switchport.tap.encapsulation.gre.strip is arista.avd.defined(true) %}
+{%             for protocol in ethernet_interface.switchport.tap.encapsulation.gre.protocols | arista.avd.natural_sort('protocol') %}
+{%                 set tap_encapsulation_cli = "switchport tap encapsulation gre protocol " ~ protocol.protocol %}
+{%                 if protocol.feature_header_length is arista.avd.defined %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " feature header length " ~ protocol.feature_header_length ~ " " ~ "strip" %}
+{%                     if protocol.re_encapsulation_ethernet_header is arista.avd.defined(true) %}
+{%                         set tap_encapsulation_cli = tap_encapsulation_cli ~ " re-encapsulation ethernet" %}
+{%                     endif %}
+{%                 else %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " strip" %}
+{%                 endif %}
+   {{ tap_encapsulation_cli }}
+{%             endfor %}
+{%             if ethernet_interface.switchport.tap.encapsulation.gre.protocols is not arista.avd.defined and
```

Comment:
Even the protocols is defined we can still have the `switchport tap encapsulation gre strip` command in EOS.
```
interface Ethernet2
   description P2P_LINK_TO_S1-SPINE1_Ethernet2
   mtu 1500
   no switchport
   ip address 172.30.255.1/31
   switchport tap encapsulation gre protocol 0xffff strip re-encapsulation ethernet
   switchport tap encapsulation gre strip
```

Reply:
Fixed